### ISSUE TITLE
integrate OpenSCAP into OSSEC

### ISSUE THREAD

**Issue by aremai** [ossec/ossec-hids#664]
would it be possible to integrate OpenSCAP into OSSEC's rootcheck module?
the way the rootcheck module is currently working is pretty static (only a plain text file).
OpenSCAP has apparently a lot of CIS benchmark tests already implemented and also offers some reporting in a OVAL and HTML format. I think it looks pretty good.

Resources I've found:
https://github.com/OpenSCAP/scap-security-guide
https://fedorahosted.org/scap-security-guide/wiki/usageguide
https://fedorahosted.org/aqueduct/

if not OpenSCAP, what else could be an alternative to the current rootcheck functionality?

**Comment by aremai** [comment]
this is meant to be a *feature request*

**Comment by atomicturtle** [comment]
Absolutely, this is a fantastic idea. Marked this up as a FR for you. We could use volunteers/feature owner to work on this.

**Comment by aremai** [comment]
great, I had hoped that you would welcome this idea :)
I'm going to engage into OpenScap over the next few months, to check how it's working...
to tell the truth I'm not a very good coder/dev, but I'd love to collaborate to get this somehow integrated into OSSEC.

also how did you add the feature-request label? how do you add labels to issues in general?

**Comment by aremai** [comment]
and the only problem that I'm seeing right now, I think OpenScap currently only works for Linux systems.
I don't know/think this would work on other *nix platforms (Solaris, AIX) but I think it would still be a good idea...

**Comment by aremai** [comment]
also another good resource:
https://github.com/OpenSCAP/openscap

**Comment by atomicturtle** [comment]
If you look on the right side of the page you'll see a "Labels" where you can add them. Or at least I can as a project administrator.   We're very grateful for whatever you can contribute here, even if you're just prototyping a feature or fix.

**Comment by spookerlabs** [comment]
Good resources here too http://www.open-scap.org/page/Main_Page
Blog from one of developer http://martin.preisler.me/category/scap/

**Comment by aremai** [comment]
ok, I will look into this properly.... maybe I can come up with an "idea" how it would work with ossec.
again I'm not a dev, but i'm intrigued to see how openscap is working.

thanks @spookerlabs for providing additional information :)

**Comment by ddpbsd** [comment]
I don't think this possibly being linux only is an issue.

Having a quick and easy "how this could work" type of post would be great.

**Comment by aremai** [comment]
great, i will study the official documentation, do some testing and hopefully get back to you with a "how this could work" draft...

**Comment by isimluk** [comment]
@aremal, Thanks for suggesting this.

OpenSCAP scanner works well on Solaris (even shipped by Oracle). We should be able to compile on AIX, but this needs further testing.

Speaking for OpenSCAP will are definitely interested to help you design the integration. Let us know any question at open-scap-list@redhat.com

**Comment by aremai** [comment]
This issue can be closed now since WAZUH have neatly integrated OpenSCAP into their OSSEC-fork.
All available here in their brand-new 2.0 release.... for more information head over here:
https://github.com/wazuh/wazuh

it not only includes OpenSCAP profiles and policies but also separate OSSEC rules and decoders to be able to detect those events properly. Since they also include the elastic stack in their Wazuh release, it also includes nice OpenSCAP dashboards for Kibana.
The Wazuh release definitely has more features and appeal to me... definitely worth checking out!

**Comment by ddpbsd** [comment]
Work done in other projects doesn't change the fact that this is still requested.

**Comment by omps** [comment]
Yup wazuh is a self package, how we do if we can do it in with the existing ELK stack.

**Comment by DominiqueDevinci** [comment]
Hello, i'am working on a project which aims to listen for system events which can be linked to a SCAP rule in order to trigger the related rule only when this event happens. Ex: if the rule is about the conf of /etc/ssh/sshd_config, the rule check is automatically triggered by the daemon (which currently use watchdog API). This technic doesn't totally exclude a periodical scan but have a good betterment since the detection is immediate, and I think it can be very great addition: in summary, a scap integration but with enhancement compared to other solutions which are all non real-time (to my knowledge at least).

What do you think of this feature enhancement proposal ? Anyway i'm in for an oscap integration, and if this idea seems also great to you ... (and as a first step i will of course start with a "classical" integration, so a periodical scan like wazuh does for instance).

Cheers,
Dominique.
